tests/test_frame.py

```diff
@@ -1180,3 +1180,72 @@ def test_columns_mixlist() -> None:
     key: List[Union[int, str]]
     key = [1]
     check(assert_type(df[key], pd.DataFrame), pd.DataFrame)
+
+
+def test_frame_scalars_slice() -> None:
+    # GH 133
+    # scalars:
+    # str, bytes, datetime.date, datetime.datetime, datetime.timedelta, bool, int,
+    # float, complex, Timestamp, Timedelta
+
+    str_ = "a"
+    bytes_ = b"1"
+    date = datetime.date(1999, 12, 31)
+    datetime_ = datetime.datetime(1999, 12, 31)
+    timedelta = datetime.datetime(2000, 1, 1) - datetime.datetime(1999, 12, 31)
+    bool_ = True
+    int_ = 1
+    float_ = 1.0
+    complex_ = 1.0 + 3.0j
+    timestamp = pd.Timestamp(0)
+    pd_timedelta = pd.Timedelta(0, unit="D")
+    none = None
+    idx = [
+        str_,
+        bytes_,
+        date,
+        datetime_,
+        timedelta,
+        bool_,
+        int_,
+        float_,
+        complex_,
+        timestamp,
+        pd_timedelta,
+        none,
+    ]
+    values = np.arange(len(idx))[:, None] + np.arange(len(idx))
+    df = pd.DataFrame(values, columns=idx, index=idx)
+
+    # Note: bool_ cannot be tested since the index is object and pandas does not
+    # support boolean access using loc except when the index is boolean
+    assert_type(df.loc[str_], pd.Series)
```

Comment:
Can you add the `check` function to all of the tests you added, so this test would look like this:
```python
check(assert_type(df.loc[str_], pd.Series), pd.Series)
```

Reply:
I added these.  They highlighted that these type defs are fundamentally wrong since scalar (only) selection using loc produces `Union[Series, DataFrame]` always.  Here is an example:

```python

import pandas as pd
df = pd.DataFrame({1:[1,2,3,4],b"1":[1,2,3,4],True:[1,2,3,4],1.0:[1,2,3,4]}, index=[1,b"1",1.0,True])
df.loc[1]
```

returns
```
      1  b'1'
1     1     1
1.0   3     3
True  4     4
```

The equality test must not be exact.  Possibly a bug in pandas.  

This said, I think the pattern here covers the common cast:

Another "gotcha" is 

```python
import numpy as np

df = pd.DataFrame(np.arange(9).reshape((3,-1)))
df.columns = [1,2,1]
df.index=[1,2,1]
df.loc[1]
```

which returns

```
   1  2  1
1  0  1  2
1  6  7  8
```

and 

```python
df[:, 1]
```

which returns

```
   1  1
1  0  2
2  3  5
1  6  8
```

Reply:
In your first example, the values `[1, 1.0, True]` are equal to 1 in python:
```python
>>> [(1 == a) for a in [1, 1.0, True]]
[True, True, True]
```
So no bug in pandas.

In your second example, the issue here is that if you have duplicate values in an axis (either the index or the columns), then that changes the type of the result.   

The goal of these stubs is to handle the most common use cases (no duplicates!).  We're doing static typing, but when the results are dynamic based on the underlying values of the `DataFrame` object, we take a best guess as to the "usual" result.

Reply:
I was under the impression that `DataFrame`s were more dict-like in that 1, 1.0, and True are all distinct keys.

Reply:
That's not true for dicts:
```python
>>> d={1: 10, 1.0: 20, True: 30}
>>> d
{1: 30}
```

Reply:
I thought it is more like `is` when it came to key distinction.  Interesting, and makes things internally consistent then. Thanks.